docs/src/tutorials/write-my-software-management-plugin.md

```diff
@@ -0,0 +1,437 @@
+sudo /etc/tedge/sm-plugins/docker remove mosquitto --module-version 1.5.7-1+deb10u1
+```
+
+Then, the plugin executes
+
+```shell
+docker rmi mosquitto:1.5.7-1+deb10u1
+```
+
+## Update-list 
+
+The `update-list` command accepts a list of software modules and associated operations as `install` or `remove`.
+This basically achieves the same purpose as original commands install and remove,
+but gets passed all software modules to be processed in one command.
+This can be needed when an order of processing software modules is relevant.
+
+In other words, you can choose a combination of the `install` or `remove` commands or this `update-list` command up to your requirement.
+If you don't want to use `update-list`, the plugin must return `1` like the first _docker_ plugin example.
+
+```shell
+case "$COMMAND" in
+    ...
+    update-list)
+        exit 1
+        ;;
+esac
+```
+
+Let's expand the first _docker_ plugin example to use `update-list`.
+First, learn what is the input of `update-list`.
+
+The Software Management Agent calls a plugin as below:
+
+```shell
+$ sudo /etc/tedge/sm-plugins/docker update-list <<EOF
+install name1 version1
+install name2 "" path2
+remove "name 3" version3
+remove name4
+EOF
+```
+
+The point is that it doesn't take any command-line argument, 
+but the software action list is sent through **stdin**.
+
+The behaviour of operations `install` and `remove` is the same as for original commands `install` and `remove`. 
+The above input is equivalent to the use of original commands (`install` and `remove`):
+
+```shell
+$ /etc/tedge/sm-plugins/docker install name1 --module-version version1
+$ /etc/tedge/sm-plugins/docker install name2 --file path2
+$ /etc/tedge/sm-plugins/docker remove "name 3" --module-version version3
+$ /etc/tedge/sm-plugins/docker remove name4
+```
+
+To make the _docker_ plugin accept a list of install and remove actions,
+let's change the file as below.
+Note that this example works only in bash.
+
+Filename: /etc/tedge/sm-plugins/docker
+
+```shell
+#!/bin/bash
+
+COMMAND="$1"
+
+read_module() {
+    if [ $# -lt 2 ]; then
+        echo "Missing version or path for sw-module '${1}'"
+        exit 1
+    elif [ $# -eq 2 ]; then
+        mOperation="$1"
+        mName="$2"
+        case "$mOperation" in
+            install)
+                docker pull $mName || exit 2
+                ;;
+            remove)
+                docker rmi $mName || exit 2
+                ;;
+        esac
+    else
+        mOperation="$1"
+        mName="$2"
+        mVersion="$3"
+        case "$mOperation" in
+            install)
+                docker pull $mName:$mVersion || exit 2
+                ;;
+            remove)
+                docker rmi $mName:$mVersion || exit 2
+                ;;
+        esac
+    fi
+}
+
+case "$COMMAND" in
+    list)
+        docker image list --format '{"name":"{{.Repository}}","version":"{{.Tag}}"}' || exit 2
+        ;;
+    install)
+        # We use update-list instead
```

Comment:
I would

* either `exit 1` to tell this sub-command is not implemented
* or provide an implementation.